acceptance/tests/resource/package/common_package_name_in_different_providers.rb

```diff
@@ -0,0 +1,107 @@
+test_name "ticket 1073: common package name in two different providers should be allowed"
+
+hosts_to_test = agents.select do |agent|
+  agent['platform'].match /(?:centos|el-|fedora)/
+end
+skip_test "No suitable hosts found" if hosts_to_test.empty?
+
+require 'puppet/acceptance/rpm_util'
+extend Puppet::Acceptance::RpmUtils
+
+rpm_options = {:pkg => 'guid', :version => '1.0'}
+
+teardown do
+  step "cleanup"
+  agents.each do |agent|
+    clean_rpm agent, rpm_options
+  end
+end
+
+def verify_state(hosts, pkg, state, match)
+  hosts.each do |agent|
+    on agent, puppet('resource', 'package', pkg, '--param provider') do
+      assert_match(%r@package\s*\{\s*'#{pkg}':\s*ensure\s*=>\s*'#{state}',\s*provider\s*=>\s*'yum',\s*\}@m, stdout)
```

Comment:
Why verify state for one package using puppet and for the other package using the underlying tool? Offhand, I'd think we'd use the underlying tool for each (since this is the 'did puppet do the right thing?' verification).